### DISCUSSION TITLE
Several Issues with firmware, chooser, and state estimates. Any Help Appreciated

### DISCUSSION THREAD

**Discussion by Ometz** [USC-ACTLab/crazyswarm#737] [Q&A]
Host of Issues with firmware, chooser, and state estimates. Any Help Appreciated.

Setup:
Razer Blade 15 Laptop running Ubuntu 20.04 on VMware Workstation 16
2x Crazyflie 2.1 with LPS decks and Flowdecks installed 
8 LPS Nodes in a 2 square meter room using TDoA3
Channel 100, designations 2 and 6 for dual deck CF’s 
Firmware on everything is the latest from the Bitcraze CFclient

Note:
Crazyflies work fine with cfclient and cf python lib from bitcraze On bitcraze VM. Note that I was unable to get cfclient installed on my own VM where crazyswarm is. (Everything is grayed out when I open it.)

Discussions I’ve already read trying to solve this:
https://github.com/USC-ACTLab/crazyswarm/issues/105
https://github.com/USC-ACTLab/crazyswarm/issues/137
https://github.com/USC-ACTLab/crazyswarm/discussions/727

What’s going on:
I have had a series of problems with Crazyswarm. It initially worked upon installation but almost immeditely ceased to work afterwards. In chronological order things progressed as follows:

Installed crazyflie in a catkin workspace (I’m dumb sorry). Was working more than right now however. Hoverswarm ran. However running hello world the CF’s failed to take off properly and instead flipped over or sideways crashing down to the ground or flew straight sideways towards the wall (or in one case straight towards my head)

After Going back to the Bitcraze VM and making sure everything still worked there I returned to my normal VM with Crazyswarm. No luck however, still had the same issue.

At somepoint crazyswarm stopped being able to connect. Running chooser.py, I kept getting timeout errors. (see figure 1) Also note that not all CF’s that I put into allcrazyflies.yaml showed up in the chooser gui but it would still show up in timeout errors as seen in the figures.

Eventually did rm -r to my crazyswarm installation in the workspace and installed it in my home directory with the intent of doing the symbolic link. No changes in behavior after doing this.

Ran again yesterday while redoing the crazyradio config steps and briefly got chooser to show battery but still could not get it to work consistently or flash firmware as seen in the figures. Additionally as is seen it cannot find the firmware to flash even during the brief moments things are working. 

Switching to the PID controller definitely did not solve my control problem. I ran the hello world script again with the PID controller selected in hoverswarm. Two things happened. Just like how CF6 is appearing in chooser gui but CF2 is not, the same is true for the Hoverswarm visualization. Nevertheless helloworld selected CF2 and proceeded to gain horizontal speed and slam into the wall at high speed ignoring my commands to kill the program (See figures 4 and 5). At this point I am totally at a loss on what to do.

Any help is appreciated!! 

Important notes:

CFClient on bitcraze VM says the link quality is adequate, usually above half full. I don’t think that is the issue. Though as seen in figures link quality has low warnings occasionally during Hoverswarm
Initially I thought maybe I need to run chooser while hoverswarm isn’t running, however even this fails often.
All figures are from today and yesterday. 

![For_CrazyGitPost1_V2](https://user-images.githubusercontent.com/12114988/236283373-6a224d7d-d922-4055-828f-7e086b740d93.PNG)
Figure 1: Timeout Error on Every Chooser Button Shown. 
It can be seen that the firmware can also not be found

![For_CrazyGitPost_2](https://user-images.githubusercontent.com/12114988/236283444-dfe45658-2c41-4271-a1a3-9010780b7a3f.PNG)
Figure 2: Trying to run list params

![For_CrazyGitPost_3](https://user-images.githubusercontent.com/12114988/236283542-6de64980-647f-4c14-9252-e520495a50e2.PNG)
Figure 3: Trying to run chooser again after sourcing. Still cannot find firmware 🙁
Notice after closing hoverswarm that at least I can get the battery voltage, this isn’t consistent however. It fails moments later. Still cannot flash firmware. 

![AFTERSLAM](https://user-images.githubusercontent.com/12114988/236284246-66daeb70-8d07-427a-9d39-91acb773b578.PNG)
Figure 4: Rviz Running, occasional link quality warnings. More happened later with the crash

![Confirmation_it_flew_CF2](https://user-images.githubusercontent.com/12114988/236284091-df182105-6824-4eea-a982-da89e0dbf7ac.PNG)
Figure 5: Crash! wouldn’t land when I wanted it to

**Comment by knmcguire** [comment]
Hi! I'm not too familiar with Crazyswarm1 itself.. but I know that support for that will be a lot less as the maintainers have moved on (as you noticed). Would it be possible for you to start with crazyswarm2: https://github.com/IMRCLab/crazyswarm2 ?

**Reply by Ometz** [reply]
Hi,

Thank you for your response. I wasn't able to solve these issues.

I have moved into Crazyswarm2 now and done some testing. In Crazyswarm2 my issues at least seemed to be narrowed down the LPS estimates. I intend to follow up more on the bitcraze discussion board now. If there is a way to just set the LPS nodes to get an initial global position and have a flowdeck take over from there. I think that would solve all of my problems given what I am trying to achieve.

Thanks
